### PR TITLE
feat: add crank to rust tests

### PR DESCRIPTION
**Pull request by ananas-block**
Changes:
- renamed InsertTwoLeavesParallel -> AppendLeaves
- refactored the indexed array queue to a queue array
- added nullify leaves instruction to ACP
- added nullify method to MockIndexer

Notes:
- this is not secure yet
- overwriting nullified leaves in queue array is not tested

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,4 +23,4 @@ ark-serialize = { git = "https://github.com/Lightprotocol/algebra", branch="rele
 overflow-checks = true
 
 [profile.test]
-opt-level = 2
+opt-level = 0
```
Comment (vadorovsky) [inline Cargo.toml:26]: Just curious - why? Is it related to the stack memory issues?

File: js/stateless.js/src/idls/account_compression.ts [hunk 2]
```diff
@@ -494,7 +555,12 @@ export type AccountCompression = {
           {
             name: 'indexedArray';
             type: {
-              array: ['u8', 192008];
+              array: [
+                {
+                  defined: 'QueueArrayElemenet';
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/idls/account_compression.ts:560]: typo should be 
```suggestion
                  defined: 'QueueArrayElement';
```

**Review by SwenSchaeferjohann — APPROVED** [review]
lgtm prolly makes sense to fix the typo after the rust tests run faster

File: Cargo.toml (hunk 1 above)
Comment (ananas-block) [reply Cargo.toml:26]: tried this for possible compiler bug

**Comment by SwenSchaeferjohann** [comment]
let's merge #520 first

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**